Src/FluentAssertions/Primitives/ObjectAssertions.cs

```diff
@@ -16,6 +16,94 @@ public ObjectAssertions(object value)
         : base(value)
     {
     }
+
+    /// <summary>
+    /// Asserts that a <typeparamref name="TExpectation"/> equals another <typeparamref name="TExpectation"/> using its <see cref="object.Equals(object)" /> implementation.
+    /// </summary>
+    /// <param name="expected">The expected value</param>
+    /// <param name="comparer">
+    /// An equality comparer to compare values.
+    /// </param>/// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndConstraint<ObjectAssertions> Be<TExpectation>(TExpectation expected, IEqualityComparer<TExpectation> comparer, string because = "", params object[] becauseArgs)
+    {
+        Guard.ThrowIfArgumentIsNull(comparer);
+
+        Execute.Assertion
+            .BecauseOf(because, becauseArgs)
+            .ForCondition(Subject is TExpectation subject && comparer.Equals(subject, expected))
+            .WithDefaultIdentifier(Identifier)
+            .FailWith("Expected {context} to be {0}{reason}, but found {1}.", expected,
```

Comment:
🤔 I would also expect some reference to the comparer, should I not?

Reply:
You mean something like including `comparer.GetType()`?

We do something similar when using an `IEqualityComparer<T>` in equivalency assertions.
https://github.com/fluentassertions/fluentassertions/pull/1284/files#diff-f060184f7198367f57bfcf2bbfc8ce3ddc5ce5db1949c93e376d6d3c780c53b8R37